processing.tex

```diff
@@ -1,5 +1,127 @@
+  To resolve this issue of unattempted quanta, we ran rescue workflows where quanta for different detectors were no longer clustered with each other.
+  Subsequently the default behavior was changed so that an individual failure among clustered quanta would not stop the remaining quanta from being attempted.
+
+\end{itemize} %step1
+
+\item step2
+\begin{itemize}
+
+ \item
+ We expected that step2 would need a total of 3 workflows ($\approx$~6600 visits each), based on quantum graph generation tests, but execution butler creation became the bottleneck instead, and we instead required 14 smaller workflows ($\approx$~1500 visits each).
+ For example, for the first of these smaller step2 workflows, quantum graph generation took 23 min, but execution butler creation took 7 hours, while job compute time was 2.5 hours (wall clock).
+ Before step3 processing started, this issue was resolved in DM-33345, and execution butler generation time was dramatically improved, e.g., from 7 hours down to 20 minutes in this step2 example, or from 1 hour down to 6.5 minutes for a 1-tract step3 workflow.
+
+\end{itemize} %step2
+
+\item step3
+\begin{itemize}
+
+  \item
+  Large numbers of simultaneous assembleCoadd (= image coaddition) jobs led to ``too many request'' errors reading from the object store (PREOPS-1034) and revealed that data caching was being done inefficiently.
+  This also led to ``hidden'' errors in the coadds, where coadd images were reported ok but actually had problems (DM-33786).
+  The first 7 step3 workflows needed to be redone due to this problem.
+  The initial fix for this caching issue was too aggressive and led to out-of-disk-space errors for a couple of large faro tasks (matchCatalogsPatchMultiBand and matchCatalogsTract), but subsequently caching was better optimized to be sufficient for coadds, but not so much to cause disk space problems for the faro tasks.
+
+  \item
+  Long-running forcedPhotCoadd jobs were failing/re-attempting at a much higher rate than before, caused by a change in the PanDA pilot version, which reset the job heartbeat timeout limit back to its default of 2 hours, whereas it had been set to 20 hours previously.
+  This issue also uncovered the need for more frequent heartbeat message logging for long-running forcedPhotCoadd (DM-33854) and faro tasks (DM-33820).
+  The timeout limit was set again to 20 hours to allow step3 production to continue efficiently.
+
+  \item
+  About once every two workflows or so, a deblend job or two failed due to out-of-memory error (DM-33690), because of very bright/large objects on the coadd image.
+  These failures were fixed in rescue workflows, but required long run times ($>$10 hrs) and large memory ($\approx$~40 GB) using the \texttt{extra-highmem} queue, though ultimately with deblending results that were likely unreliable.
```

Comment:
We also had problems with pre-emption didn't we and had to switch to a non-pre-emptible queue.

Reply:
Yes, this was needed for last obstinate deblend quantum, and this is mentioned in the next line in the text.